api/src/main/java/org/apache/iceberg/expressions/InclusiveMetricsEvaluator.java

```diff
@@ -372,6 +372,38 @@ public Boolean or(Boolean leftResult, Boolean rightResult) {
       return ROWS_MIGHT_MATCH;
     }
 
+    @Override
+    public <T> Boolean notStartsWith(BoundReference<T> ref, Literal<T> lit) {
+      Integer id = ref.fieldId();
+
+      if (containsNullsOnly(id)) {
+        return ROWS_CANNOT_MATCH;
+      }
+
+      ByteBuffer prefixAsBytes = lit.toByteBuffer();
+
+      Comparator<ByteBuffer> comparator = Comparators.unsignedBytes();
+
+      if (lowerBounds != null && upperBounds != null &&
+              lowerBounds.containsKey(id) && upperBounds.containsKey(id)) {
+        ByteBuffer lower = lowerBounds.get(id);
```

Comment:
I did not add a `null` check on lower, as I didn't see it in any of the other metrics based tests. But I thought I'd bring it up as this will definitely NPE if it's possible for the column to be in the `upperBounds` and `lowerBounds` map with a null value. Seems like that shouldn't happen though.

Reply:
Given it's not checked elsewhere, I'm resolving this comment to make the PR more digestible to others.